public/app/features/alerting/unified/components/contact-points/constants.ts

```diff
@@ -1,3 +1,28 @@
+import { AccessControlAction } from 'app/types';
+
 export const RECEIVER_STATUS_KEY = Symbol('receiver_status');
 export const RECEIVER_META_KEY = Symbol('receiver_metadata');
 export const RECEIVER_PLUGIN_META_KEY = Symbol('receiver_plugin_metadata');
+
+/**
+ * List of granular permissions that allow viewing contact points
+ *
+ * Any permission in this list will be checked for client side access to view Contact Points functionality.
+ */
+const PERMISSIONS_CONTACT_POINTS_READ = [AccessControlAction.AlertingReceiversRead];
+
+/**
+ * List of granular permissions that allow modifying contact points
+ */
+export const PERMISSIONS_CONTACT_POINTS_MODIFY = [
+  AccessControlAction.AlertingReceiversCreate,
+  AccessControlAction.AlertingReceiversWrite,
+];
+
+/**
+ * List of all permissions that allow contact points read/write functionality
+ *
+ * Any permission in this list will also be checked for whether the built-in Grafana Alertmanager is shown
+ * (as the implication is that if they have one of these permissions, then they should be able to see Grafana AM in the AM selector)
+ */
+export const PERMISSIONS_CONTACT_POINTS = [...PERMISSIONS_CONTACT_POINTS_READ, ...PERMISSIONS_CONTACT_POINTS_MODIFY];
```

Comment:
How about moving this to a file like `/contact-points/permissions`? We could also move the showManageContactPointPermissions function from the utils file there. I think keeping related functionality together in dedicated files would make things easier to maintain.

Based on my experience, files like `misc` and `utils` tend to become obsolete quickly and lead to code duplication over time. (no one really cares about these files)

Reply:
:+1: Have moved this to a `permissions` file